Better separate application and library code

jacobdeery opened the pull request:
Move all application code into a new top level "application" directory,
and put the "resources" folder into that directory as well so that we
can use it to store resources Qt needs (like the icon).

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/waterloo-rocketry/operations-simulator/3)
<!-- Reviewable:end -->

jacobdeery committed:
Better separate application and library code

Move all application code into a new top level "application" directory,
and put the "resources" folder into that directory as well so that we
can use it to store resources Qt needs (like the icon).

wendi-yu reviewed (approved):
Yep looks good
> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/operations-simulator/3)* status: :shipit: complete! all files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->

jacobdeery committed:
Fix bug when run as a Python script

getattr needs a default value or it will AttributeError when main is run
as a standalone script since sys won't have the 'frozen' attribute.

wendi-yu reviewed (approved):
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/operations-simulator/3)* status: :shipit: complete! all files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->
